frontend/ui/widget/imageviewer.lua

```diff
@@ -781,25 +776,10 @@ function ImageViewer:onSaveImageView()
         self:update()
         UIManager:forceRePaint()
     end
-    Screen:shot(screenshot_name)
-    local widget = ConfirmBox:new{
-        text = T( _("Saved screenshot to %1.\nWould you like to set it as screensaver?"), BD.filepath(screenshot_name)),
-        ok_text = _("Yes"),
-        ok_callback = function()
-            G_reader_settings:saveSetting("screensaver_type", "image_file")
-            G_reader_settings:saveSetting("screensaver_image", screenshot_name)
-            if restore_settings_func then
-                restore_settings_func()
-            end
-        end,
-        cancel_text = _("No"),
-        cancel_callback = function()
-            if restore_settings_func then
-                restore_settings_func()
-            end
-        end
-    }
-    UIManager:show(widget)
+    UIManager:sendEvent(Event:new("Screenshot"))
+    if restore_settings_func then
+        restore_settings_func()
+    end
```

Comment:
In the previous code, restore_settings_func was executed after the dialog is dismissed (so, we see the current image without decorations, restore_settings_func is the thing that restores the decoration).
With the new code, it's run nearly immediately, and the dialog is shown over the restored decoration I think.
(Yes, you added a "View", but that requires another step.)

(I found that neat, but I'm not especially attached to that.)

Reply:
As I understand, the purpose of removing the decoration was to make screenshot of the full image only (I don't think it is necessary, I want to screenshot what I see, but okay, let it be).
Is there a reason to duplicate the full screenshot dialog code in ImageViewer to just have a pause to view the full image?
If we want to view the image without decoration we can just tap it.

Reply:
I would remove this saving/restoring at all.

Reply:
There's another kind of decoration you can't remove: an optional caption at top, that you get at least with 2nd++ images in Wikipedia full lookup results.
This long tap in ImageViewer is more a "save that image (with the possible zoom/crop I can make)" than a "screenshot what I see".

> Is there a reason to duplicate the full screenshot dialog code in ImageViewer to just have a pause to view the full image?

This duplication is probably more (my) lazyness than anything else :) unless there was a reason, may be this callback.
But you can still have/keep that callback.
Just pass a `when_done_func`, something like:
`UIManager:sendEvent(Event:new("Screenshot", restore_settings_func))`
`function Screenshoter:onScreenshot(filename, when_done_func)` that you would call, if provided, in all callbacks.